Contributing.md

```diff
@@ -1,32 +1,47 @@
-* Request new documentation with [GitHub Issues](https://github.com/OfficeDev/office-js-docs-reference/issues).
+## Add or edit new samples
 
-## Contribute using GitHub
+> **Important**: The reference content in this repository is autogenerated. Please read the [documentation tooling notes](https://github.com/OfficeDev/office-js-docs-reference/blob/master/DocumentationToolingNotes.md) if you're planning to edit files not mentioned in this section.
 
-> **Important**: The reference content in this repository is autogenerated. Please read the [documentation tooling notes](https://github.com/OfficeDev/office-js-docs-reference/blob/master/DocumentationToolingNotes.md) before editing files.
+Code samples found in documentation from this repo come from to places:
+
+* [**/docs/code-snippets**](https://github.com/OfficeDev/office-js-docs-reference/tree/master/docs/code-snippets)
+* [OfficeDev/office-js-snippets](https://github.com/OfficeDev/office-js-snippets) (The source of Script Lab samples)
+
+Contributions to the Script Lab sample repo are covered in [that repo's readme](https://github.com/OfficeDev/office-js-snippets#readme).
+
+### How to add or edit sample code
+
+The [**/docs/code-snippets/\*-snippets.yaml**](https://github.com/OfficeDev/office-js-docs-reference/tree/master/docs/code-snippets) files in this repo contain samples for each application and the Common APIs. Each YAML key in one of those YAML files is the UID of a field in the documentation. The keys follow the format `<Application>.<ClassName>#<fieldName>:member<overloadNumber(methods only)>`. Add your sample code snippet as the value under the key of the field you're using in a sample. Use the following Git mechanisms to submit changes to the repo.
+
+* To make small changes to a sample, [Contribute using GitHub](#contribute-using-github).
+* To make large changes, [Contribute using Git](#contribute-using-git).
+
+### Tips on making good samples
+
+We'll review each PR to make sure it meets Microsoft standards. To help ensure your sample gets integrated into the reference documentation, keep the following tips in mind.
+
+* Limit the scope. Sample code that focuses on a small scenario or API usage helps readers understand how that API is used. (Larger samples should go to [Script Lab](https://github.com/OfficeDev/office-js-snippets) or the [Code Samples repo](https://github.com/OfficeDev/office-js-snippets).)
+* Include a brief comment explaining the intended behavior.
+* Clearly state the expected input or state.
+
+### Contribute using GitHub
 
 Use GitHub to contribute to this documentation without having to clone the repo to your desktop. This is the easiest way to create a pull request in this repository. Use this method to make a minor change that doesn't involve code changes.
 
 Using this method allows you to contribute to one article at a time.
 
-### To Contribute using GitHub
+#### To Contribute using GitHub
```

Comment:
This procedure appears to apply to editing a reference article, but don't you really have to create an issue request since any changes must be made at DefinitelyTyped?

Reply:
The intent here is as a follow-up to the "How to add or edit sample code" section. Those *-snippets files can and should be edited this way.